botocore/utils.py

```diff
@@ -1569,6 +1571,45 @@ def set_endpoint(self, request, **kwargs):
     def _use_accesspoint_endpoint(self, request):
         return 's3_accesspoint' in request.context
 
+    def _validate_fips_supported(self, request):
+        client_region = request.context.get('client_region', '')
+        if 'fips' not in client_region:
+            return
+        if 'outpost_name' in request.context['s3_accesspoint']:
+            raise UnsupportedS3AccesspointConfigurationError(
+                msg=(
+                    'Invalid configuration Outpost Access Points '
+                    'do not support Fips- regions'
+                )
+            )
+        client_region = client_region.replace('fips-', '')
+        if request.context['s3_accesspoint']['region'] != client_region:
+            if self._s3_config.get('use_arn_region'):
+                raise UnsupportedS3AccesspointConfigurationError(
+                    msg=(
+                        'Invalid configuration, '
+                        'FIPS region does not match ARN region'
+                    )
+                )
+            else:
+                raise UnsupportedS3AccesspointConfigurationError(
+                    msg=(
+                        'Invalid configuration, cross region Access Point ARN'
+                    )
+                )
+
+    def _validate_global_regions(self, request):
+        if self._s3_config.get('use_arn_region'):
```

Comment:
I'd assume we would want to use `self._s3_config.get('use_arn_region', True)` here since by default we use the arn region? Otherwise, commands like this no longer work as they used to resolve to the us-east-1 endpoint:
```
$ aws s3api get-object --bucket arn:aws:s3:us-east-1:123456789012:accesspoint:myendpoint --key foo --region s3-external-1 /tmp/foo

Unsupported configuration when using S3 access-points: Invalid configuration, client region is not a regional endpoint
```

Reply:
I'm not sure. Looks like here we use opposite logic https://github.com/boto/botocore/blob/92c63ea94542c5d54d582c50509be02b96d7a443/botocore/utils.py#L1811

Reply:
I double checked this, the default is `True` for the S3 client, but for the S3 control client it looks like it defaults to `False`. I think we need to stick with the default for the S3 client. Mainly if we made the default `False` here, we would introduce a regression where we would now fast fail if:
1. The user has `AWS_S3_US_EAST_1_REGIONAL_ENDPOINT=regional` set (In v2, this environment variable no longer exists and essentially defaults to the `regional` value).
2. The user does not set a region.

```
$ AWS_S3_US_EAST_1_REGIONAL_ENDPOINT=regional aws s3api get-object --bucket arn:aws:s3:us-east-1:123456789012:accesspoint:myendpoint --key foo  /tmp/foo
Unsupported configuration when using S3 access-points: Client is configured to use the global psuedo-region "aws-global". When providing access-point ARNs a regional endpoint must be specified.
```
I pushed up the update in the latest commit to fix this